app/plex_auto_collections.py

```diff
@@ -64,12 +151,58 @@ def update_from_config(config_path, plex, headless=False, no_meta=False, no_imag
 
         tmdb_id = None
         actor_id = None
-        methods = [m for m in collections[c] if m not in ("tmdb_biography", "tmdb_profile", "sync_mode", "tmdb-list", "imdb-list", "trakt-list", "tmdb-poster", "details", "subfilters", "sort_title", "content_rating", "summary", "tmdb_summary", "collection_mode", "collection_sort", "poster", "tmdb_poster", "file_poster", "background", "file_background", "name_mapping")]
+        actor_method = None
+        methods = [m for m in collections[c] if m not in details]
+        def alias(filter, alias):
+            if filter[-1] == "!":
+                return alias[filter[:-1]] + "!"
+            elif filter[-4] == ".lte":
+                return alias[filter[:-4]] + ".lte"
+            elif filter[-4] == ".gte":
+                return alias[filter[:-4]] + ".gte"
+            else:
+                return alias[filter]
         subfilters = []
         if "subfilters" in collections[c]:
-            for sf in collections[c]["subfilters"]:
-                sf_string = sf, collections[c]["subfilters"][sf]
-                subfilters.append(sf_string)
+            if collections[c]["subfilters"]:
+                for sf in collections[c]["subfilters"]:
+                    try:
+                        final_sf = alias(sf, method_alias)
+                        print("| Config Warning: {} subfilter will run as {}".format(sf, final_sf))
+                    except KeyError:
+                        final_sf = sf
+                    if final_sf in movie_only_subfilters and libtype == "show":
+                        print("| Config Error: {} subfilter only works for movie libraries".format(final_sf))
+                    elif final_sf in all_subfilters:
+                        sf_string = final_sf, collections[c]["subfilters"][sf]
+                        subfilters.append(sf_string)
+                    else:
+                        print("| Config Error: {} subfilter not supported".format(sf))
+            else:
+                print("| Config Error: subfilters attribute is blank")
+        and_filters = []
```

Comment:
I'm not in love with the name `and_filter`, but this is good stuff.

Reply:
So, just taking a step back - I _really_ like the functionality here, but I'm afraid it might be too confusing. What if we simplified it to just two attributes:
1. "Simple" Plex filters the way they are now - `genre`, `studio`, etc., where multiple filter `or` together simply. I'd be fine if we wanted to create an attribute `plex_filter` - seems like it might simplify the code too.
2. "Custom" filters, where there is an attribute `custom_filter` that is an ordered dictionary or filters and `custom_filter_logic` that can rearrange the filters. I'd be fine if we call this `custom_plex_filter` and `custom_plex_filter_logic` too - seems like it might be more readable. For example

```yaml
Good Romance or Romantic Comedies:
  custom_filter:
    - genre: romance               # 1
    - genre: comedy                # 2
    - rating.gte: 7                # 3
  custom_filter_logic: ({1} or ({1} and {2})) and {3}
```

Feels like we could borrow Tautulli's method here to make this easy

https://github.com/Tautulli/Tautulli/blob/14b98a32e085d969f010f0249c3d2f660db50880/plexpy/helpers.py#L925-L1007

Reply:
looking at this it looks really nice but I think it would be hard to combine subfilters and filters like that. Subfliters affect all movies in the collection including whats added via other lists/attributes where the original filters are just a one time search. Another problem with using `.and` is that we can't `.and.not`  so we would need another method thats maybe `.ant`. 

Lets not mix the subfilters with the filters so what we need to decide is what to do about `.and` personally my vote would be to use `.and` and `.ant` i like the commands being all the same length at the end

Reply:
oh wait lets prefix it with `and.` and then it wont get in the way of not and we wont need another identifier

i like it, i think it looks nice like that

```yaml
collections:
  Dave Chappelle:
    actor: Dave Chappelle
    and.genre: Comedy
```

Reply:
The documentation of filters vs. subfilters is great and even more thorough than the `plexapi`, but I agree that it's still potentially confusing. I think we should stick to the original structure until we can nail down how to simplify it to something like @burkasaurusrex posted with "simple" and "complex/custom/advanced" filters. This is closest to how it's presented in the Plex UI. It does mean that filters and subfilters are mixed and would complicate how things are passed to the `plexapi`. I think we should also normalize the filter names to whatever is in the Plex UI, such as using `release_date` vs. `originally_available`. Custom things like `max_age`, we can name ourselves. 

For now, let's just add these features:
- `max_age`
- `.gte` and `.lte`